src/main/java/org/bigml/binding/localmodel/Tree.java

```diff
@@ -975,26 +975,31 @@ public Prediction predict(final JSONObject inputData, List<String> path,
                 	confidence = regressionError(unbiasedSampleVariance(distribution, prediction),
                             totalInstances, DEFAULT_RZ);
                 }
-                
+
                 Integer dMin = !this.regression ? null : this.min;
                 Integer dMax = !this.regression ? null : this.max;
-                
+
                 return new Prediction(prediction, confidence, totalInstances,
                         distributionMedian(distribution, totalInstances),
                         path, distribution, distributionUnit,
-                        lastNode.getTree().getChildren(), 
+                        lastNode.getTree().getChildren(),
                         dMin,
                         dMax);
             } else {
                 JSONArray distribution  = Utils.convertDistributionMapToSortedArray(finalDistribution);
                 long totalInstances = calculateTotalInstances(distribution);
 
-                return new Prediction(((JSONArray) distribution.get(0)).get(0),
-                        wsConfidence(((JSONArray) distribution.get(0)).get(0), distribution,
-                                totalInstances, DEFAULT_RZ),
-                        totalInstances, null,
-                        path, distribution, "categorical",
-                        lastNode.getTree().getChildren(), null, null);
+                if (distribution.size() > 0) {
+                    return new Prediction(((JSONArray) distribution.get(0)).get(0),
```

Comment:
move line 990 inside this if?

Reply:
Yes, and 989 too. I'll update the PR.

Reply:
989 ???

Reply:
Of course not, what fool would suggest that? ... Me!  Never mind. Just 990!